### PR TITLE
Support OIDC authentication flags

### LINKED ISSUE #190: Allow passing flags for OIDC
I've seen an attempt to use OIDC with KCP in https://www.youtube.com/watch?v=FD_kY3Ey2pI (incomplete). I'm thinking to create a demo based on that feature, so it would be nice if kcp could accept oidc/rbac related flags on startup without having to hack on code.

Is this something planned? I could also help.

### PR DESCRIPTION
**Pull request by imjasonh**
Fixes #190

### CODE REVIEW

File: pkg/server/config.go
Comment (stevekuznetsov) [inline pkg/server/config.go:10]: Isn't this going to be a disallowed import?

File: pkg/server/config.go
Comment (stevekuznetsov) [inline pkg/server/config.go:61]: This code gets removed in #228 so I wouldn't try *too* hard to refactor if you have anything else planned ;)

Comment (imjasonh) [reply pkg/server/config.go:61]: Excellent. I'll wait for that then. This was all pretty annoying because this seemed to want to be set up using [`AddFlags`](https://github.com/kubernetes/kubernetes/blob/63e7ee43bb5394abd6dd5010be47ebcb22ec0ef8/pkg/kubeapiserver/options/authentication.go#L245) but the way we added and parsed flags as two separate steps made that harder. I bet your change will make this all easier 👍

File: pkg/server/config.go
Comment (ncdc) [reply pkg/server/config.go:10]: It's fine for now, given that all of kcp is in the prototyping phase. But if we want to do this long-term/"for real", we need to move anything we need in k/k/pkg into k/k/staging & publish it as a standalone repo (or move it into a preexisting staging dir/repo).

**Comment by stevekuznetsov** [comment]
Given Andy's comment, this LGTM - after a rebase we can merge

**Comment by stevekuznetsov** [comment]
@imjasonh based on the e2e, the `kcp` binary does not start

**Comment by imjasonh** [comment]
> @imjasonh based on the e2e, the `kcp` binary does not start

This was caused by nil clientcert config because I was only populating `WithOIDC`. I've temporarily switched to `WithAll` and parsing alllllll the authentication flags, but I'm not sure we want them all. We can probably just only define/parse the OIDC flags manually, if we feel strongly about it.

**Review by stevekuznetsov — APPROVED** [review]

**Comment by davidfestal** [comment]
Does anyone know how to trigger the run of the missing checks that seem to prevent the merging ?

**Comment by stevekuznetsov** [comment]
Wow, "re-run" does not run the updated ones .... @imjasonh mind re-pushing?

**Pull request merged**
